sweep/sweeper.go

```diff
@@ -1651,6 +1582,85 @@ func (s *UtxoSweeper) ListSweeps() ([]chainhash.Hash, error) {
 	return s.cfg.Store.ListSweeps()
 }
 
+// handleNewInput processes a new input by registering spend notification and
+// scheduling sweeping for it.
+func (s *UtxoSweeper) handleNewInput(input *sweepInputMessage,
+	bestHeight int32) {
+
+	outpoint := *input.input.OutPoint()
+	pendInput, pending := s.pendingInputs[outpoint]
+	if pending {
+		log.Debugf("Already pending input %v received", outpoint)
+
+		s.handleExistingInput(input, pendInput)
+
+		return
+	}
+
+	// Create a new pendingInput and initialize the listeners slice with
+	// the passed in result channel. If this input is offered for sweep
+	// again, the result channel will be appended to this slice.
+	pendInput = &pendingInput{
+		listeners:        []chan Result{input.resultChan},
+		Input:            input.input,
+		minPublishHeight: bestHeight,
+		params:           input.params,
+	}
+	s.pendingInputs[outpoint] = pendInput
+	log.Tracef("input %v added to pendingInputs", outpoint)
+
+	// Start watching for spend of this input, either by us or the remote
+	// party.
+	cancel, err := s.waitForSpend(
+		outpoint, input.input.SignDesc().Output.PkScript,
+		input.input.HeightHint(),
+	)
+	if err != nil {
+		err := fmt.Errorf("wait for spend: %v", err)
+		s.signalAndRemove(&outpoint, Result{Err: err})
+
+		return
+	}
+
+	pendInput.ntfnRegCancel = cancel
+
+	// Check to see if with this new input a sweep tx can be formed.
+	if err := s.scheduleSweep(bestHeight); err != nil {
+		log.Errorf("schedule sweep: %v", err)
+	}
+
+	log.Tracef("input %v scheduled", outpoint)
+}
+
+func (s *UtxoSweeper) handleExistingInput(input *sweepInputMessage,
```

Comment:
Missing `godoc` comment.

Reply:
updated!

Reply:
> Or is there another usecase already ?

Yes! If you look at how we update the params via `BumpFee` and how we handle existing inputs, it's clear the logic can be merged, something to add maybe in the following PR?

Reply:
Great, yes let's do it in the follow-up PR